### PR TITLE
Implement support for cookies with JSON values

### PR DESCRIPTION
**Pull request by jthomerson**

### CODE REVIEW

File: src/Request.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default class Request {
     * Contains cookies sent by the request (key/value pairs). If the request contains no
     * cookies, it defaults to `{}`.
     */
-   public readonly cookies: StringMap;
+   public readonly cookies: { [key: string]: any };
```
Comment (yokuze) [inline src/Request.ts]: ADIRL - we should change `any` to `unknown`.

File: src/Request.ts [hunk 2]
```diff
@@ -351,14 +351,26 @@ export default class Request {
       }, {});
    }
 
-   private _parseCookies(): StringMap {
+   private _parseCookies(): { [key: string]: any } {
```
Comment (yokuze) [inline src/Request.ts]: ADIRL - we should change `any` to `unknown`.

File: src/Response.ts [hunk 3]
```diff
@@ -219,13 +219,21 @@ export default class Response {
...
+    * @see https://github.com/expressjs/express/blob/dc538f6e810bd462c98ee7e6aae24c64d4b1da93/lib/response.js#L836-L838
+    *
     * TODO: how does a user see the documentation for `CookieOpts`?
     *
     * @param name the name of the cookie
     * @param userVal the value of the cookie
     * @param userOpts the options (such as domain, path, etc)
     */
-   public cookie(name: string, userVal: string, userOpts?: CookieOpts): Response {
+   public cookie(name: string, userVal: any, userOpts?: CookieOpts): Response {
```
Comment (yokuze) [inline src/Response.ts]: ADIRL - we should change `any` to `unknown`.

**Review by yokuze — CHANGES_REQUESTED** [review]

**Comment by yokuze** [comment]
Just the comments about `unknown` - everything else looks good.

**Comment by jthomerson** [comment]
@yokuze updated with all the unknowns 😆

**Review by yokuze — APPROVED** [review]
Looks good!

**Pull request merged**